usr/lib/icsf_stdll/icsf.c

```diff
@@ -2154,7 +2155,8 @@ static int icsf_ber_decode_get_attribute_list(BerElement * berbuf,
                 }
                 attrs[i].ulValueLen = attrbval.bv_len;
             } else {
-                rc = CKR_BUFFER_TOO_SMALL;
+                rc = 8;
+                *reason = 3003; /* CKR_BUFFER_TOO_SMALL */
```

Comment:
lgtm.
If the number of different ICSF errors is growing, we could think of introducing constants for the ICSF reason codes.